ambry-tools/src/main/java/com.github.ambry/tools/admin/ConcurrencyTestTool.java

```diff
@@ -0,0 +1,975 @@
+      byte[] usermetadata = new byte[new Random().nextInt(1024)];
+      BlobProperties props = new BlobProperties(randomNum, "test");
+      try {
+        final long startTime = SystemTime.getInstance().nanoseconds();
+        ByteBufferReadableStreamChannel putChannel = new ByteBufferReadableStreamChannel(ByteBuffer.wrap(blob));
+        router.putBlob(props, usermetadata, putChannel, new Callback<String>() {
+          @Override
+          public void onCompletion(String result, Exception exception) {
+            long latencyPerBlob = SystemTime.getInstance().nanoseconds() - startTime;
+            logger.trace(" Time taken to put blob id " + result + " in ms " + latencyPerBlob / SystemTime.NsPerMs
+                + " for blob of size " + blob.length);
+            Exception exceptionToReturn = null;
+            Pair<String, byte[]> toReturn = null;
+            if (result != null) {
+              toReturn = new Pair(result, blob);
+            } else {
+              exceptionToReturn = exception;
+            }
+            futureResult.done(toReturn, exceptionToReturn);
+            if (callback != null) {
+              callback.onCompletion(toReturn, exceptionToReturn);
+            }
+          }
+        });
+      } catch (Exception e) {
+        futureResult.done(null, e);
+        if (callback != null) {
+          callback.onCompletion(null, e);
+        }
+      }
+    }
+
+    /**
+     * {@inheritDoc}
+     * Request to fetch the blob from the server directly pertaining to the {@code blobId} passed in and verify for
+     * its content
+     * @param blobIdStr the (string representation of the) blobId of that blob that needs to be fetched
+     * @param blobContent the content of the blob that needs to be verified against
+     * @param expectedErrorCode the expected error code for the Get call
+     * @param futureResult the {@link FutureResult} where the result/exception will be set on completion
+     * @param callback the {@link Callback} that will be called on completion
+     */
+    @Override
+    public void getBlobAndValidate(final String blobIdStr, final byte[] blobContent,
+        final RouterErrorCode expectedErrorCode, final FutureResult futureResult, final Callback callback) {
+      try {
+        final Long startTimeGetBlob = SystemTime.getInstance().nanoseconds();
+        router.getBlob(blobIdStr, new GetBlobOptions(), new Callback<GetBlobResult>() {
+          @Override
+          public void onCompletion(GetBlobResult getBlobResult, Exception exception) {
+            long latencyPerBlob = SystemTime.getInstance().nanoseconds() - startTimeGetBlob;
+            logger.trace("Time taken to get blob " + blobIdStr + " in ms " + latencyPerBlob / SystemTime.NsPerMs);
+            Exception exceptionToReturn = null;
+            if (getBlobResult != null) {
+              logger.trace("OnCompletion returned for blob " + blobIdStr + " with non null result ");
+              try {
```

Comment:
It seems there is no need to have this `try-catch` block, all error cases are handled within `CompareRunnable`.